python/pyiceberg/serializers.py

```diff
@@ -14,27 +14,72 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+from __future__ import annotations
 
 import codecs
+import gzip
 import json
+from abc import ABC, abstractmethod
+from typing import Callable
 
 from pyiceberg.io import InputFile, InputStream, OutputFile
 from pyiceberg.table.metadata import TableMetadata, TableMetadataUtil
 
+GZIP = "gzip"
+
+
+class Compressor(ABC):
+    @staticmethod
+    def get_compressor(location: str) -> Compressor:
+        return GzipCompressor() if location.endswith(".gz.metadata.json") else NOOP_COMPRESSOR
+
+    @abstractmethod
+    def get_stream_decompressor(self, inp: InputStream) -> InputStream:
+        return inp
+
+    @abstractmethod
+    def get_bytes_compressor(self) -> Callable[[bytes], bytes]:
+        pass
+
+
+class NoopCompressor(Compressor):
+    def get_stream_decompressor(self, inp: InputStream) -> InputStream:
+        return inp
+
+    def get_bytes_compressor(self) -> Callable[[bytes], bytes]:
```

Comment:
Is `get` adding any value in this API? I generally omit it or replace it with a more specific verb.

Reply:
Thanks, for of habit!